openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TestCaseRepository.java

```diff
@@ -233,7 +229,33 @@ public RestUtil.PutResponse<?> addTestCaseResult(
     return new RestUtil.PutResponse<>(Response.Status.CREATED, changeEvent, RestUtil.ENTITY_FIELDS_CHANGED);
   }
 
-  public RestUtil.PutResponse<?> deleteTestCaseResult(String updatedBy, String fqn, Long timestamp) {
+  private TestCaseResolutionStatus setTestCaseFailureStatus(TestCase testCase, String updatedBy) {
+    String json =
+        daoCollection
+            .testCaseResolutionStatusTimeSeriesDao()
+            .getLatestExtension(testCase.getFullyQualifiedName(), TESTCASE_RESULT_EXTENSION);
```

Comment:
We can improve this query by looking for non-resolved test case resolution status only. 
I can imagine if the test case going through its life cycle there may be multiple statuses here and we may pull all of them in this query?